src/printer.js

```diff
@@ -613,21 +688,154 @@ function printStatement(node) {
     case "useitem":
       return node.name;
     case "retif":
-    case "eval":
+      return group(
+        concat([
+          printNode(node.test),
+          indent(
+            concat([
+              line,
+              "? ",
+              printNode(node.trueExpr),
+              line,
+              ": ",
+              printNode(node.falseExpr)
+            ])
+          )
+        ])
+      );
     case "exit":
-    case "halt":
+      return concat(["exit(", printNode(node.status), ")"]);
     case "clone":
-    case "declare":
+      return concat(["clone ", printNode(node.what)]);
+    case "declare": {
+      const printDeclareArguments = function(node) {
+        const directive = Object.keys(node.what)[0];
+        return concat([directive, "=", printNode(node.what[directive])]);
+      };
+      const printDeclareChildren = function(node) {
+        return concat(
+          node.children.map(child =>
+            concat([
+              printNode(child),
+              lineShouldEndWithSemicolon(child) ? ";" : ""
+            ])
+          )
+        );
+      };
+      if (node.mode === "short") {
+        return concat([
+          "declare(",
+          printDeclareArguments(node),
+          "):",
+          hardline,
+          printDeclareChildren(node),
+          hardline,
+          "enddeclare;"
+        ]);
+      } else if (node.mode === "block") {
+        return concat([
+          "declare(",
+          printDeclareArguments(node),
+          ") {",
+          indent(concat([hardline, printDeclareChildren(node)])),
+          hardline,
+          "}"
+        ]);
+      }
+      return concat([
+        "declare(",
+        printDeclareArguments(node),
+        ");",
+        hardline,
+        printDeclareChildren(node)
+      ]);
+    }
     case "global":
+      return group(
+        concat([
+          "global",
+          indent(
+            concat([line, join(concat([",", line]), node.items.map(printNode))])
+          )
+        ])
+      );
     case "static":
+      return group(
+        concat([
+          "static",
+          indent(
+            concat([
+              line,
+              join(
+                concat([",", line]),
+                node.items.map(item => {
+                  // @TODO: hacking this for now. assignments nested inside a static
+                  // declaration doesn't have the operator set, so printing manually
```

Comment:
Would be nice to send issues/prs on the parser implementation. Prettier has found a ton of edge cases on all the parsers we used and having a good relationship with them was critical to make the project work.